### PR TITLE
first cut of changes for authorization endpoint live

### PR DESCRIPTION
**Pull request by maheshp**
**WIP do not merge**

### CODE REVIEW

File: source/includes/authorization/_033_fetch_access_token.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## Fetch Access Token
+
+This is a message that a web based plugin should implement. In case of web based authentication a user is authenticated by an external authorization server, on successful authentication the authorization server should redirect to GoCD endpoint returned by [authorization server redirect url] request, we would refer to these requests as pre-authenticated requests. Depending upon the authorization server the pre-authenticated requests can contain user information either in HTTP header or request params. GoCD upon receiving a pre-authenticated request would make the fetch access token request to the plugin by passing all the user information provide by the authorization server, the plugin in turn should exchange this information with the authorization server to request for an access token to make subsequent requests on behalf of the user.
```
Comment (ketan) [inline source/includes/authorization/_033_fetch_access_token.md]: `authorization server redirect url` — did you intend to add a link?

File: source/includes/authorization/_033_fetch_access_token.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+<p class='attributes-table-follows'></p>
+
+| Key            | Type     | Description                                                                   |
+|----------------|----------|-------------------------------------------------------------------------------|
+| `auth_configs` | `Object` | This key contains list of `<authconfig>` configured for the plugin. |
+| `role_configs` | `Object` | This key contains list of `<roleconfig>` configured for the plugin. |
+
+<p class='request-body-heading'>Request parameters</p>
+Request paramters would contain all request parameters sent by the external authorization server to GoCD.
```
Comment (ketan) [inline source/includes/authorization/_033_fetch_access_token.md]: Line break before?

File: source/includes/authorization/_033_fetch_access_token.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+<p class='request-body-heading'>Request headers</p>
+Request headers would contain all the HTTP request headers sent by the external authorization server to GoCD.
```
Comment (ketan) [inline source/includes/authorization/_033_fetch_access_token.md]: Line break before?

File: source/includes/authorization/_033_fetch_access_token.md (hunk 1 above)
Comment (maheshp) [inline source/includes/authorization/_033_fetch_access_token.md]: yes, will add

**Comment by jyotisingh** [comment]
`can_authorize` capability needs to be documented. missing now

File: source/includes/authorization/_033_fetch_access_token.md (hunk 2 above)
Comment (maheshp) [inline source/includes/authorization/_033_fetch_access_token.md]: done

File: source/includes/authorization/_033_fetch_access_token.md (hunk 3 above)
Comment (maheshp) [inline source/includes/authorization/_033_fetch_access_token.md]: done

**Pull request merged**